test/parallel/xyk-pallet.multiswap.errors.test.ts

```diff
@@ -0,0 +1,184 @@
+    }
+  });
+  it.each([0])(
+    "[gasless] disabled on token of the chained polls",
+    async (position: number) => {
+      await Assets.disableToken(tokenIds[position]);
+      const testUser1 = users[0];
+      let exception = false;
+      await expect(
+        multiSwapBuy(testUser1, tokenIds, new BN(1000), BN_TEN_THOUSAND).catch(
+          (reason) => {
+            exception = true;
+            throw new Error(reason.data);
+          }
+        )
+      ).rejects.toThrow(
+        "1010: Invalid Transaction: The swap prevalidation has failed"
+      );
+      expect(exception).toBeTruthy();
+    }
+  );
+  it.each([2, 4])(
+    "[gasless] disabled on token of the chained polls - Fail on tx execution-%s",
+    async (position: number) => {
+      await Assets.disableToken(tokenIds[position]);
+      const testUser1 = users[0];
+      const multiSwapOutput = await multiSwapBuy(
+        testUser1,
+        tokenIds,
+        new BN(1000),
+        BN_TEN_THOUSAND
+      );
+
+      const eventResponse = getEventResultFromMangataTx(multiSwapOutput, [
+        "xyk",
+        "MultiBuyAssetFailedOnAtomicSwap",
+      ]);
+      expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+
+      const boughtTokens = await getUserBalanceOfToken(
+        tokenIds[tokenIds.length - 1],
+        testUser1
+      );
+      expect(boughtTokens.free).bnEqual(BN_ZERO);
+    }
+  );
+});
+describe("Multiswap - error cases: pool status & gasless integration", () => {
+  beforeAll(async () => {
+    try {
+      getApi();
+    } catch (e) {
+      await initApi();
+    }
+    await setupApi();
+    api = getApi();
+    ({ users, tokenIds } = await setup5PoolsChained(users));
+  });
+  test("[gasless] High value swaps are disabled on multiswap", async () => {
+    const testUser0 = users[0];
+    const meta = await api.query.feeLock.feeLockMetadata();
+    const threshold = new BN(
+      JSON.parse(JSON.stringify(meta)).swapValueThreshold as any as BN
+    );
+    const feeLockAmount = new BN(
+      JSON.parse(JSON.stringify(meta)).feeLockAmount as any as BN
+    );
+    let tokenList = tokenIds.concat(MGA_ASSET_ID);
+    tokenList = tokenList.reverse();
+    testUser0.addAssets(tokenList);
+    await testUser0.refreshAmounts(AssetWallet.BEFORE);
+
+    await multiSwapSell(testUser0, tokenList, threshold.addn(10));
+
+    await testUser0.refreshAmounts(AssetWallet.AFTER);
+    const diff = testUser0.getWalletDifferences();
+    expect(
+      diff.find((x) => x.currencyId === MGA_ASSET_ID)?.diff.reserved
+    ).bnEqual(feeLockAmount);
+  });
+  test("[gasless] Fail on client when not enough MGAs to lock AND tokens that exist whitelist", async () => {
+    const keyring = new Keyring({ type: "sr25519" });
+    const testUser1 = new User(keyring);
+    await Sudo.batchAsSudoFinalized(Assets.mintToken(tokenIds[0], testUser1));
+    const meta = await api.query.feeLock.feeLockMetadata();
+    //TODO:Update whitelist!
```

Comment:
Is that leftover or tokens in this test are not whitelisted?

Reply:
you are right, I forgot it.  will do that now!